pkg/provider/provider.go

```diff
@@ -63,6 +63,24 @@ func Provider(version string) *schema.Provider {
 				Description: "The default region if not specified in the resource",
 				DefaultFunc: schema.EnvDefaultFunc("MZ_DEFAULT_REGION", "aws/us-east-1"),
 			},
+			"host": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The Materialize host. Can also come from the `MZ_HOST` environment variable.",
+				DefaultFunc: schema.EnvDefaultFunc("MZ_HOST", nil),
+			},
+			"port": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The Materialize port. Can also come from the `MZ_PORT` environment variable.",
+				DefaultFunc: schema.EnvDefaultFunc("MZ_PORT", 6875),
```

Comment:
For self-hosted, do we want to point at the external SQL address, or the internal one? The external one doesn't allow login as `mz_system`, and other users have limited permissions.

Also, nitpick, it might be good to name this in a way that is clearly a SQL port, as we also have HTTP(S) ports.